rust/src/treefile.rs

```diff
@@ -511,22 +501,33 @@ impl Treefile {
         }
     }
 
-    fn update_checksum(&self, repo: &ostree::Repo, checksum: &mut glib::Checksum) -> Result<()> {
+    fn get_checksum(&self, repo: &ostree::Repo) -> Result<String> {
```

Comment:
I got a bit lost around here.
It looks like this (and the `.as_str()` a few lines below) is moving from digesting bytes to handling strings (Rust UTF-8 or C nul-terminated? I'm unsure).
I'd naively expected only byte-slices to be involved here, without any String/GString/Cstring involvement.
Can we get rid of string-conversions in this function, and return a plain `Result<Vec<u8>>`?

Reply:
I added the `.as_str()` mainly because it was useful for dropping a `dbg!(rev)` in there - the debug display for GString isn't useful and `dbg!()` on `&[u8]` is unreadable.

We are casting the `&str` to bytes when hashing.  As far as returning a `Vec<u8>` for output...dunno, it doesn't matter much whether we hash the hex digest or the raw bytes I don't think.

Reply:
> It looks like this (and the .as_str() a few lines below) is moving from digesting bytes to handling strings (Rust UTF-8 or C nul-terminated? I'm unsure).

To clarify, this hasn't actually changed.  A [GString](http://gtk-rs.org/docs/glib/struct.GString.html) here is basically a NUL terminated string we got from C that we are asserting is UTF-8 (GLib convention is that's what `char*` means, if you want non-UTF-8 you use `guint8*/GBytes/etc`).

Reply:
Thanks for the feedback, that's exactly where I stumbled! Now I went down to `ostree_commit_get_content_checksum` and just noticed that it returns an hex-encoded checksum, which explain the logic in this consumer.
I think I got tricked by the `(guint8*)` cast here below, sorry for the detour.

Reply:
Yeah it's confusing; one already has to keep in mind all of Rust's various string/bytestring types and when interacting with C it gets even more confusing with `CStr` and then `GString` is its own thing that also encompasses the "owned by C", just borrowed case.